consoles/console.pm

```diff
@@ -85,6 +86,29 @@ sub select {
     return $activated;
 }
 
+sub save_snapshot {
+    my ($self, $name) = @_;
+    $self->snapshot($name, 'activated', $self->{activated});
+}
+
+sub load_snapshot {
+    my ($self, $name) = @_;
+    $self->{activated} = $self->snapshot($name, 'activated') // 0;
+}
+
+sub snapshot {
+    my ($self, $name, $key, $value) = @_;
+    return undef unless defined($name);
+
+    if (defined($key) && defined($value)) {
+        $self->{snapshots}->{$name} //= {};
```

Comment:
you should actually not need that, see https://perlmaven.com/autovivification . Just try to access hash entries without specifying the intermediate layers

Reply:
Thx for the hint, wasn't aware of autovivification!